### ISSUE TITLE
LTeX doesn't work properly when mother tongue is set to `fr-FR`

### ISSUE THREAD

**Issue by lrnv** [valentjn/vscode-ltex#260]
Edit: Afer some digging around, i remarked that disabling the option for the Picky rules retunrs to a stage where everything works as intended. This means that the Picky rules interfer with the automatic updates of the rules files, although it seems like ltex is still picking up every change to the files (deletions, manual changes, automatic changes...).

**Describe the bug**
The function 'disable rule' lists the rule into `.vscode/ltex.disabledRules.en-US.txt` but the rule keep showing anyway. 

**Steps to reproduce**
I've set the mother tong to french and the picky behavior. Some of the picky rules are not needed since I do not make the mistake it tracks, but i cannot disable them correctly. 
I also tried disabling a rule that is not from these picky rules, did not work (the name of the rule is written in the file, but the rule still appears).

**Expected behavior**
`Disable rule` function should... disable the rule ! ;)

**Sample document**
If the bug occurs for a specific document (e.g. LaTeX), please paste it here. If your document is very long or confidential, please create and attach a smaller example for which the bug still occurs so that we can reproduce it.

<details>

```
Disabling the WORD_CONTAINS_UNDERSCORE rule in this document does not work !
```

</details>

**LTeX configuration**
Please paste all configuration settings starting with `ltex.` from your `settings.json`. You can help us by temporarily removing some irrelevant settings from your `settings.json` and see if the bug still occurs.

<details>

```
    "ltex.additionalRules.enablePickyRules": true,
    "ltex.additionalRules.motherTongue": "fr-FR",
```

</details>

But also the ltex.disabledRules.en-US.txt : 
<details>

```
    WORD_CONTAINS_UNDERSCORE
```

</details>

**"LTeX Language Server" log file**
First, reproduce the bug. Then, go to `View` → `Output` and select `LTeX Language Server` in the drop-down list. Paste this log here:

<details>

```
PR�CIS: Checking the following text in language 'en-US' via LanguageTool: "Disabling the WORD_CONTAINS_UNDERSCORE rule in this document does not work !\n"
f�vr. 25, 2021 4:19:51 PM org.bsplines.ltexls.server.DocumentChecker checkAnnotatedTextFragment
PR�CIS: Obtained 1 rule match
```

</details>

**"LTeX Language Client" log file**
First, set the `ltex.trace.server` setting in your `settings.json` to `"verbose"`. Then, reload the VS Code window and reproduce the bug. Go to `View` → `Output` and select `LTeX Language Client` in the drop-down list. Paste this log here (note: it will contain your checked document):

<details>

```
REPLACE_THIS_WITH_LTEX_LANGUAGE_CLIENT_LOG
```

</details>

**Version information**
List here the version information of the relevant software.

- Operating system: Windows 10
- VS Code: 1.54.0-insider
- vscode-ltex: 9.0.0

**Additional context/information**
No additional information, the same bug appears whatever the file. It's annoying :/

**Comment by lrnv** [comment]
It seems like the problem comes from the fact that i inputted `fr-FR` as a mothertong instead of just `fr`. Putting `fr` solved the problem.

**Comment by Ezibenroc** [comment]
I had a similar problem, adding a word to the dictionary did not change anything, the word was still underlined. Changing the mothertong setting from `fr-FR` to `fr` also solved this.

Maybe this should somehow be fixed? I do not think this is the intended behavior.

**Comment by valentjn** [comment]
Thanks for the feedback. While trying to reproduce this, it turned out that `fr-FR` isn't actually a proper language code that LanguageTool recognizes. The possible codes are the same as for [`ltex.language`](https://valentjn.github.io/vscode-ltex/docs/settings.html#ltexlanguage), so `fr` is the correct one (I'm not sure why LanguageTool's language codes are so inconsistent). This was a bug in the config spec of LT<sub>E</sub>X, which allowed arbitrary strings for [`ltex.additionalRules.motherTongue`](https://valentjn.github.io/vscode-ltex/docs/settings.html#ltexadditionalrulesmothertongue). I changed this to an `enum` of all possible values. I'm not sure about the behavior when setting this to `fr-FR`, though. It seemed to be non-deterministic to me, and I don't think it's related to picky rules. It's probably related to cached (previous) instances of the LanguageTool checker and/or of checked sentences. Maybe the treatment of errors can be improved.

**Comment by lrnv** [comment]
Making this an Enum is indeed probably the right decision. 

Maybe we could tell LanguageTool that the langauges codes they use are inconsistant and non-standard (I do not know if there exists a standard though).

**Comment by valentjn** [comment]
Fix released in 10.0.0.
